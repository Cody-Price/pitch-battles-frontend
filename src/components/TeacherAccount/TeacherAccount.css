.teacher-account {
  position: relative;
}

.teacher-logout-label {
  position: absolute;
  right: 1rem;
  top: -1rem;
}

.teacher-logout-label:hover {
  cursor: pointer;
}
