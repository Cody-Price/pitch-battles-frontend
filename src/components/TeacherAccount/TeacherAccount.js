import React, { Component } from "react";
import "./TeacherAccount.css";
import PropTypes from "prop-types";
import EditProfile from "../EditProfile/EditProfile";
import ChangePassword from "../ChangePassword/ChangePassword";

class TeacherAccount extends Component {
  render() {
    console.log(this.props.webToken);
    return (
      <main className="teacher-account">
        <p onClick={this.props.logout} className="teacher-logout-label">
          logout
        </p>
        <h3 className="teacher-account-label">Teacher Account Info</h3>
        <EditProfile
          userType="teacher"
          user={this.props.user}
          changeProfile={this.props.changeProfile}
          getUpdatedUserData={this.props.getUpdatedUserData}
          webToken={this.props.webToken}
        />
        <h3 className="teacher-password-label">Change Password</h3>
        <ChangePassword
          userType="teacher"
          user={this.props.user}
          updateWebToken={this.props.updateWebToken}
          webToken={this.props.webToken}
        />
      </main>
    );
  }
}

export default TeacherAccount;


TeacherAccount.propTypes = {
  logout: PropTypes.func,
  user: PropTypes.object,
  webToken: 'string',
  updateWebToken: PropTypes.func
  changeProfile: PropTypes.func
};