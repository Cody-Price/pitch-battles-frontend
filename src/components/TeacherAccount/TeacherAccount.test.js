import React from "react";
import ReactDOM from "react-dom";
import TeacherAccount from "./TeacherAccount";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("TeacherAccount", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TeacherAccount />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  
});
