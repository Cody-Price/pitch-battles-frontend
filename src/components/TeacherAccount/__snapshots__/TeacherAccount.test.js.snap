// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherAccount should match the snapshot 1`] = `
<main
  className="teacher-account"
>
  <p
    className="teacher-logout-label"
  >
    logout
  </p>
  <h3
    className="teacher-account-label"
  >
    Teacher Account Info
  </h3>
  <EditProfile
    userType="teacher"
  />
  <h3
    className="teacher-password-label"
  >
    Change Password
  </h3>
  <ChangePassword
    userType="teacher"
  />
</main>
`;
